frontend/javascripts/oxalis/controller/segment_mesh_controller.ts

```diff
@@ -84,80 +102,106 @@ export default class SegmentMeshController {
     scale: Vector3 | null = null,
     lod: number,
     layerName: string,
+    additionalCoordinates: AdditionalCoordinate[] | null | undefined,
   ): void {
-    if (this.meshesGroupsPerSegmentationId[layerName] == null) {
-      this.meshesGroupsPerSegmentationId[layerName] = {};
-    }
-    if (this.meshesGroupsPerSegmentationId[layerName][segmentationId] == null) {
-      this.meshesGroupsPerSegmentationId[layerName][segmentationId] = {};
+    const additionalCoordinatesString = getAdditionalCoordinatesAsString(additionalCoordinates);
+    const newGroup = new THREE.Group();
+    const keys = [additionalCoordinatesString, layerName, segmentationId, lod];
+    _.set(
+      this.meshesGroupsPerSegmentationId,
+      keys,
+      _.get(this.meshesGroupsPerSegmentationId, keys, newGroup),
+    );
+    if (lod === NO_LOD_MESH_INDEX) {
+      this.meshesLODRootGroup.addNoLODSupportedMesh(newGroup);
+    } else {
+      this.meshesLODRootGroup.addLODMesh(newGroup, lod);
     }
-    if (this.meshesGroupsPerSegmentationId[layerName][segmentationId][lod] == null) {
-      const newGroup = new THREE.Group();
-      this.meshesGroupsPerSegmentationId[layerName][segmentationId][lod] = newGroup;
-      if (lod === NO_LOD_MESH_INDEX) {
-        this.meshesLODRootGroup.addNoLODSupportedMesh(newGroup);
-      } else {
-        this.meshesLODRootGroup.addLODMesh(newGroup, lod);
-      }
-      // @ts-ignore
-      newGroup.cellId = segmentationId;
-      if (scale != null) {
-        newGroup.scale.copy(new THREE.Vector3(...scale));
-      }
+    // @ts-ignore
+    newGroup.cellId = segmentationId;
+    if (scale != null) {
+      newGroup.scale.copy(new THREE.Vector3(...scale));
     }
     const mesh = this.constructMesh(segmentationId, geometry);
     if (offset) {
       mesh.translateX(offset[0]);
       mesh.translateY(offset[1]);
       mesh.translateZ(offset[2]);
     }
-
-    this.meshesGroupsPerSegmentationId[layerName][segmentationId][lod].add(mesh);
+    this.addMeshToMeshGroups(additionalCoordinatesString, layerName, segmentationId, lod, mesh);
   }
 
-  removeMeshById(segmentationId: number, layerName: string): void {
-    if (this.meshesGroupsPerSegmentationId[layerName] == null) {
-      return;
-    }
-    if (this.meshesGroupsPerSegmentationId[layerName][segmentationId] == null) {
-      return;
+  removeMeshById(
+    segmentationId: number,
+    layerName: string,
+    additionalCoordinates: AdditionalCoordinate[] | null | undefined,
+  ): void {
+    // either remove a mesh for specific additional coordinates, or remove all meshes for a segment per default by passing null as additionalCoordinates.
+    let additionalCoordinatesToRemoveMeshes;
```

Comment:
this is unnecessary because this method is only ever called with the current add. coords -> remove param and get addCoords in method